internal/healthcheck/healthcheck_test.go

```diff
@@ -0,0 +1,687 @@
+					MachineCount:      1,
+					ReadyMachineCount: 12,
+				}},
+				&configv1.ClusterVersion{
+					Status: configv1.ClusterVersionStatus{
+						Conditions: []configv1.ClusterOperatorStatusCondition{
+							{
+								Type:   configv1.OperatorAvailable,
+								Status: configv1.ConditionFalse,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "happy path",
+			wantErr: false,
+			objects: []runtime.Object{
+				&configv1.Infrastructure{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cluster",
+					},
+					Status: configv1.InfrastructureStatus{
+						InfrastructureTopology: configv1.SingleReplicaTopologyMode,
+					},
+				},
+				&v1.Node{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							NodeRoleControlPlane: "",
+							NodeRoleMaster:       "",
+							NodeRoleWorker:       "",
+						},
+					},
+					Status: v1.NodeStatus{
+						Conditions: []v1.NodeCondition{
+							{
+								Type:   v1.NodeReady,
+								Status: v1.ConditionTrue,
+							},
+							{
+								Type:   v1.NodeNetworkUnavailable,
+								Status: v1.ConditionFalse,
+							},
+						},
+					},
+				},
+				&operatorsv1alpha1.ClusterServiceVersion{
+					Status: operatorsv1alpha1.ClusterServiceVersionStatus{
+						Phase:  operatorsv1alpha1.CSVPhaseSucceeded,
+						Reason: operatorsv1alpha1.CSVReasonInstallSuccessful,
+					},
+				},
+				&configv1.ClusterOperator{
+					Status: configv1.ClusterOperatorStatus{
+						Conditions: []configv1.ClusterOperatorStatusCondition{
+							{
+								Status: configv1.ConditionTrue,
+								Type:   configv1.OperatorAvailable,
+							},
+						},
+					},
+				},
+				&mcv1.MachineConfigPool{Status: mcv1.MachineConfigPoolStatus{
+					MachineCount:      1,
+					ReadyMachineCount: 1,
+				}},
+				&configv1.ClusterVersion{
+					Status: configv1.ClusterVersionStatus{
+						Conditions: []configv1.ClusterOperatorStatusCondition{
+							{
+								Type:   configv1.OperatorAvailable,
+								Status: configv1.ConditionTrue,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			// apply CRs
+			s := scheme.Scheme
+			s.AddKnownTypes(configv1.GroupVersion, &configv1.ClusterVersion{})
+			s.AddKnownTypes(configv1.GroupVersion, &configv1.ClusterVersionList{})
+			s.AddKnownTypes(mcv1.GroupVersion, &mcv1.MachineConfigPoolList{})
+			s.AddKnownTypes(mcv1.GroupVersion, &mcv1.MachineConfigPool{})
+			s.AddKnownTypes(configv1.GroupVersion, &configv1.ClusterOperatorList{})
+			s.AddKnownTypes(configv1.GroupVersion, &configv1.ClusterOperator{})
+			s.AddKnownTypes(operatorsv1alpha1.SchemeGroupVersion, &operatorsv1alpha1.ClusterServiceVersionList{})
+			s.AddKnownTypes(operatorsv1alpha1.SchemeGroupVersion, &operatorsv1alpha1.ClusterServiceVersion{})
+			s.AddKnownTypes(configv1.GroupVersion, &configv1.Infrastructure{})
+			s.AddKnownTypes(v1.SchemeGroupVersion, &v1.Node{})
```

Comment:
nit: I think these do not have to be called every time a testcase is executed. Perhaps move them out of the for loop or define them only once in an init function?

Reply:
Yep! Updated to use init()